pageserver/src/tenant/timeline/walreceiver/walreceiver_connection.rs

```diff
@@ -435,6 +435,14 @@ pub(super) trait ExpectedError {
     /// If this function returns an `false` the error should be propagated and the connection manager
     /// will log the error at ERROR level.
     fn is_expected(&self) -> bool;
+
+    /// If is_expected() is true, use this function to optionally transform the error's string
+    /// representation instead of using Display directly to get a string-ized version of the error.
+    /// It may be filtered to omit misleading "error" parts of a non-harmful message, to help
+    /// with searching logs for errors.
+    fn maybe_reformat(&self) -> Option<String> {
+        None
+    }
```

Comment:
I think returning None here will make your additions dead code, as in there's no way to run them in practice.

Ah but this is the default value. Why add one if both implementors have this? Confusing, just remove the body:

```suggestion
    fn maybe_reformat(&self) -> Option<String>;
```

Noting my dislike for `String` and I think we can avoid it. Rustful way would be to have a newtype, but I think we should avoid this and *recognize* the message we want, then just format it with a static str as mentioned in https://github.com/neondatabase/neon/pull/4902/files#r1285711379.

Reply:
This is defunct since adding WalReceiverError.